shell/helpers.sh

```
@@ -0,0 +1,113 @@
+        if [[ -x "$candidate" ]]; then
+            printf '%s\n' "$candidate"
+            return 0
+        fi
+    done
+    _dotbins_helpers_log warn "tool '$tool' not found in $bin_dir"
+    return 1
+}
+
+
+dotbins_show_missing() {
+    local bin_dir="$(_dotbins_helpers_bin_dir)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)